tests/components/dunehd/test_config_flow.py

```diff
@@ -0,0 +1,98 @@
+"""Define tests for the Dune HD config flow."""
+from homeassistant import data_entry_flow
+from homeassistant.components.dunehd.const import DOMAIN
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_USER
+from homeassistant.const import CONF_HOST
+
+from tests.async_mock import patch
+from tests.common import MockConfigEntry
+
+CONFIG_HOSTNAME = {CONF_HOST: "dunehd-host"}
+CONFIG_IP = {CONF_HOST: "10.10.10.12"}
+
+DUNEHD_STATE = {"protocol_version": "4", "player_state": "navigator"}
+
+
+async def test_import(hass):
+    """Test that the import works."""
+    with patch("pdunehd.DuneHDPlayer.update_state", return_value=DUNEHD_STATE):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=CONFIG_HOSTNAME
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == "dunehd-host"
+        assert result["data"] == {CONF_HOST: "dunehd-host"}
+
+
+async def test_import_cannot_connect(hass):
+    """Test that errors are shown when cannot connect to the host during import."""
+    with patch("pdunehd.DuneHDPlayer.update_state", return_value={}):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=CONFIG_HOSTNAME
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5633381)